dataservice/api/common/schemas.py

```diff
@@ -19,18 +24,46 @@ class Meta:
         exclude = ('uuid',)
         dump_only = ('created_at', 'modified_at')
 
+    @pre_dump(pass_many=True)
+    def wrap_pre(self, data, many):
+        if isinstance(data, Pagination):
+            self.__pagination__ = data
+            return data.items
+        return data
+
     @post_dump(pass_many=True)
     def wrap_envelope(self, data, many):
+        """
+        Wraps ORM objects in a standard response format
+
+        If `many=False`, only one object is being returned and any `_links`
+        that are in the data are moved out to the response's `_links`.
+
+        If `many=True`, many objects are being returned and the top `_links`
+        in the response will be populated with pagination details.
+        """
+        resp = {'_status': {'message': self.status_message,
+                            'code': self.status_code}}
+        # Move links to the envelope links if just a single object
         if not many and '_links' in data:
             _links = data['_links']
             del data['_links']
+        # Insert pagination object, if there is one
+        elif many and self.__pagination__ is not None:
+            p = self.__pagination__
+            _links = {}
+            _links['self'] = request.full_path
```

Comment:
Hmm, I think you might want something different here. If I send this request:  
`localhost:5000/participants?after=adasdf`, I get back:
```
{
    "_links": {
        "next": "/participants?after=1519762761.405353",
        "self": "/participants?after=adasdf"
    },
    "_status": {
        "code": 200,
        "message": "success"
    },
    "limit": 10,
    "results": [
      actual results here ...
    ],
    "total": 300
}
```
Maybe you could do something like: 
```python
if p.after.timestamp() == 0.0:
    after = None
else:
    after = p.after

_links['self'] = url_for(self.Meta.resource_url,
                         after=after)
```
Which would return: 
```
{
    "_links": {
        "next": "/participants?after=1519762761.405353",
        "self": "/participants
    },
    "_status": {
        "code": 200,
        "message": "success"
    },
    "limit": 10,
    "results": [
        actual results here ...
    ],
    "total": 300
}
```
Or maybe you could do:
```python
_links['self'] = url_for(self.Meta.resource_url,
                         after=p.after)
```
Which would keep the default of unix epoch. However, if I was the user and didn't know that this timestamp was epoch I might be a little confused as to where the random date came from 😛 . It'd be less confusing to me if i just got back `self: /participants`

Reply:
Made it so that it will default to just `/participants` if the `after` param cannot be parsed.